### PR TITLE
- Add filter data hook

### PR DESCRIPTION
**Pull request by jgololicic**
Search filter hook which takes data stream (pooling data for example) and search term string (search bar for example) and 
provides another filtered stream which reacts on any of the two input streams.

### CODE REVIEW

File: frontend/src/shared/hooks/filter-data.ts [hunk 1]
```diff
@@ -0,0 +1,13 @@
+export interface UseFilterDataHook<T> {
+    filteredData: T;
+}
+
+export const useFilterData = <T>(data: T[], search: string, props?: string[]): UseFilterDataHook<T[]> => {
+    if (!Array.isArray(data) || data.length === 0) {
+        return { filteredData: data };
+    }
+
+    props = props || Object.keys(data[0]);
```
Comment (bartolomej) [inline frontend/src/shared/hooks/filter-data.ts:10]: Maybe we should give a more suitable name to the "props" variable, feels too general ? Something like "filterByKeys" or "filterByProps" maybe ?

File: frontend/src/pages/blocks/Blocks.tsx [hunk 2]
```diff
@@ -3,27 +3,38 @@ import { useSearch } from '../../shared/hooks/search';
...
             <p>Search term: {searchTerm}</p>
             {data && <p>Data length: {data?.length}</p>}
+            {filteredData && <p>Filtered data length: {filteredData?.length}</p>}
 
             <div className={classes.listContainer}>
-                {data &&
-                    data.length &&
-                    data.map((d: any, index: number) => (
+                {filteredData &&
+                    filteredData.length &&
```
Comment (bartolomej) [inline frontend/src/pages/blocks/Blocks.tsx:30]: As I understand, `filteredData` can be only empty or non-empty array, so there is no need to check that it's "truthty" and "has length" ? Should we just map over items like: `<div> { filteredData.map(e => ... ) }  </div>` ?

**Review by bartolomej — APPROVED** [review]

File: frontend/src/shared/hooks/filter-data.ts (hunk 1 above)
Comment (jgololicic) [reply frontend/src/shared/hooks/filter-data.ts:10]: good point

File: frontend/src/pages/blocks/Blocks.tsx (hunk 2 above)
Comment (jgololicic) [reply frontend/src/pages/blocks/Blocks.tsx:30]: this part is was only done to test other functionality = crap code :)

**Pull request merged**